python/graphstorm/dataloading/metadata.py

```diff
@@ -0,0 +1,958 @@
+            'edge data information.'),
+            self.__class__.__name__)
+        return self._edges
+
+    def is_homogeneous(self):
+        """ Check if the metadata graph is a homogeneous one.
+        """
+        return self._graph_metadata.is_homogeneous()
+
+    @property
+    def ndata(self):
+        """ Provide an interface for retrieving node data in a homogeneous metadata graph.
+
+        This interface is for homogeneous graphs only, and the node type should be `_N`, which is
+        defined by dgl's DEFAULT_NTYPE constant.
+        """
+        if not self.is_homogeneous():
+            logging.warning(
+                'Only homogeneous metadata graphs support the "ndata" interface, but got a '
+                'heterogeneous metadata graph.')
+            return None
+        else:
+            return self._nodes[DEFAULT_NTYPE].data
+
+    @property
+    def edata(self):
+        """ Provide an interface for retrieving edge data in a homogeneous metadata graph.
+
+        This interface is for homogeneous graphs only, and the edge type should be
+        (`_N`, `_E`, `_N`), which is defined by dgl's `dgl.NID` and `dgl.EID` constants.
+        """
+        if not self.is_homogeneous():
+            logging.warning(
+                'Only homogeneous metadata graphs support the "edata" interface, but got a '
+                'heterogeneous metadata graph.')
+            return None
+        else:
+            return self._edges[DEFAULT_ETYPE].data
+
+
+class DglDataViewSimulation():
+    """ Simulation of the DGL NodeView and EdgeView class.
+
+    .. versionadded:: 0.5.0
+        Add ``DglDataViewSimulation`` in 0.5.0 to support real-time inference on Amazon SageMaker
+        endpoints.
+
+    This class provides a simple interface of DGL NodeView/EdgeView class, i.e., it only provides
+    the `data` interface for retrieving node/edge feature data.
+
+    The `data` property is a dictionary to host feature names with their corresponding tensors.
+    """
+    def __init__(self):
+        self._data = {}
+
+    @property
+    def data(self):
+        """ Get the data property for setting and getting
+        """
+        return self._data
+
+
+class GSDglDistGraphFromMetadata(GSDglGraphFromMetadata):
+    """ A metadata graph implementation for DGL distributed graph APIs.
+
+    .. versionadded:: 0.5.0
+        Add ``GSMetadataDglDistGraph`` in 0.5.0 to support lightweight DGL distributed graph
+        construction using graph metadata only.
+
+    This metadata graph only provides two DGL distributed graph APIs, i.e.,
+
+    - get_node_partition_policy(ntype)
+    - get_partition_book()
+
+    However, as DGL distributed graph requires actual graph structure data, which the metadata
+    cannot provide. Therefore, this metadata graph will raise NotImplementedError to let
+    callers know this metadata graph cannot be used for DGL distributed graph.
```

Comment:
Can we clarify the need for this class? It will provide the APIs that `GSDglGraphFromMetadata` provides, but why do we add two functions just to raise NotImplemented errors? Isn't it better to just not provide the functions?

Reply:
We dedicatedly added these two APIs to provide users hints that the `GSDglDistGraphFromMetedata` cannot rebuild models with learnable embeddings in the current version.